sklearn/linear_model/_ridge.py

```diff
@@ -1520,15 +1529,38 @@ def fit(self, X, y, sample_weight=None):
                     identity_estimator = _IdentityClassifier(
                         classes=np.arange(n_y)
                     )
-                    predictions_, y_ = predictions, y.argmax(axis=1)
+                    alpha_score = scorer(identity_estimator,
+                                         predictions, y.argmax(axis=1))
                 else:
                     identity_estimator = _IdentityRegressor()
-                    predictions_, y_ = predictions.ravel(), y.ravel()
-
-                alpha_score = scorer(identity_estimator, predictions_, y_)
-
-            if (best_score is None) or (alpha_score > best_score):
-                best_coef, best_score, best_alpha = c, alpha_score, alpha
+                    if self.alpha_per_target:
+                        alpha_score = np.array([
+                            scorer(identity_estimator,
+                                   predictions[:, j], y[:, j])
+                            for j in range(n_y)
+                        ])
+                    else:
+                        alpha_score = scorer(identity_estimator,
+                                             predictions.ravel(), y.ravel())
+
+            # Keep track of the best model
+            if best_score is None:
+                # initialize
+                best_coef = c
+                best_score = alpha_score
+                if self.alpha_per_target:
+                    best_alpha = np.full(n_y, alpha)
+                else:
+                    best_alpha = alpha
+            else:
+                # update
+                if self.alpha_per_target:
+                    to_update = alpha_score > best_score
+                    best_coef[:, to_update] = c[:, to_update]
+                    best_score[to_update] = alpha_score[to_update]
+                    best_alpha[to_update] = alpha
```

Comment:
The docstring for `alpha_`, `best_score_`, etc would need to be updated to reflect these new shapes.